### ISSUE TITLE
Implement approximate ion jacobian in kinetic equation (collision operator)

### ISSUE THREAD

**Issue by Embreus** [chalmersplasmatheory/DREAM#96]
We should explore the possibility of neglecting the jacobian with respect to ion densities that have negligible density when constructing the kinetic equation. This could dramatically reduce nnz (probably by at least a factor of 2) which could have a large impact on computation times.

We could for example choose to only keep the jacobian w.r.t. the N largest densitites of each ion species, or densities above some threshold. Typically, in the self-consistent IonRateEquation charge distribution, almost all but ~3-4 states are completely negligible.

**Comment by Embreus** [comment]
The affected terms are 
`SlowingDownTerm`
`PitchScatterTerm`
`EnergyDiffusionTerm`
`IonKineticIonizationTerm`
I suggest that we implement a setting (in DistributionFunction, perhaps?) where the user can set an `ionDensityThresholdForJacobian` parameter, which could be for example '0.07' if we only want charge states with density >7% of the maximum for that species to contribute to the jacobian. If set to 1, the ion jacobian would be neglected altogether.

**Comment by Embreus** [comment]
This was implemented in 196e82d. I opted to only enable/disable the entire ion jacobian in the kinetic equation, because I realize that there is a risk of creating a numerical instability when introducing a threshold:
For ion states where the jacobian is zero, the equation will think that if it reaches those the collision operator will nearly vanish (since it accounts for the jacobian in the currently occupied states), which may allow an unphysical self-consistent solution that it will aim for. 
(kind of hard to explain, and I think it is slightly far fetched, but I worry that it could become more unstable than just ignoring the jacobian altogether)

The `IonKineticIonizationTerm` implements its own approximate jacobian mode already, and the new setting doesn't influence it.
